code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -104,8 +115,28 @@
 		log_combat(user, src, "punched", "hulk powers")
 
 /obj/vehicle/sealed/mecha/blob_act(obj/structure/blob/B)
-	log_message("Attack by blob. Attacker - [B].", LOG_MECHA, color="red")
-	take_damage(30, BRUTE, MELEE, 0, get_dir(src, B))
+	if (!B?.overmind?.blobstrain)
+		return take_damage(30, BRUTE, MELEE, 0, get_dir(src, B))
+
+	var/datum/blobstrain/strain = B.overmind.blobstrain
+	log_message("Attack by blob strain: [strain.name]. Attacker - [B].", LOG_MECHA, color="red")
+
+	var/damage = 50
+
+	switch (strain.type)
```

Comment:
no need for a switch on only two checks. just looks weird (also the spaces are weird but thats my personal take)

Reply:
tysm, I just copied it from the generator haha. I was trying to add more strain interactions like acid on blazing oil and slime but after messing about for a bit I realized adding acid to mechs would be a bit more work